applications/FluidDynamicsApplication/custom_processes/shock_capturing_entropy_viscosity_process.h

```diff
@@ -0,0 +1,350 @@
+    };
+
+
+    /// Pointer definition of ShockCapturingEntropyViscosityProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ShockCapturingEntropyViscosityProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor with model
+    ShockCapturingEntropyViscosityProcess(
+        Model& rModel,
+        Parameters rParameters)
+        : ShockCapturingEntropyViscosityProcess(rModel.GetModelPart(rParameters["model_part_name"].GetString()), rParameters) {};
+
+    /// Constructor with model part
+    ShockCapturingEntropyViscosityProcess(
+        ModelPart& rModelPart,
+        Parameters rParameters);
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void ExecuteInitializeSolutionStep() override;
+
+    void ExecuteFinalizeSolutionStep() override;
+
+    int Check() override;
+
+    const Parameters GetDefaultParameters() const override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "ShockCapturingEntropyViscosityProcess";
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override { rOStream << "ShockCapturingEntropyViscosityProcess"; }
+
+    /// Print object's data.
+    void PrintData(std::ostream&) const override {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+    bool mFirstTimeStep = true;
+    bool mComputeAreasEveryStep = false;
+    double mEntropyConstant = 0.0;
+    double mEnergyConstant = 0.0;
+    double mArtificialMassDiffusivityPrandtl = 0.0;
+    double mArtificialConductivityPrandtl = 0.0;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void ValidateAndAssignParameters(Parameters &rParameters);
+
+    void UpdateNodalAreaProcess();
+
+    /**
+     * @brief Computes nodal entropies and initializes artificial variables to 0.
+     *
+     * @tparam WriteBufferIndex: The buffer index to write the result in. This is an
+     * ugly hack to deal with the first time-step.
+     */
+    template<unsigned int WriteBufferIndex = 0>
+    void ComputeNodalEntropies();
+
+    void ComputeArtificialMagnitudes();
+
+    void DistributeVariablesToNodes(
+        Element& rElement,
+        const double ArtificialDynamicViscosity,
+        const double ArtificialBulkViscosity,
+        const double ArtificialConductivity) const;
+
+    static double ComputeEntropy(const double Density, const double Pressure, const double Gamma);
```

Comment:
```suggestion
    static inline double ComputeEntropy(
        const double Density,
        const double Pressure,
        const double Gamma);
```

Reply:
I don't think `inline` is an appropiate keyword here.

If I understand correctly, you want this function to be inlined by the compiler. However, this keyword does not compel the compiler to inline the function. See here [StackOverflow](https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method) or here [YouTube](https://www.youtube.com/watch?v=GldFtXZkgYo).

Reply:
Agree on the formatting change though

Reply:
You want this function to be `inline`. The compiler is right now most probably doing it. The `inline` is sort of a suggestion just in case but is the compiler the one taking the final decision.